vyper/signatures/interface.py

```diff
@@ -85,18 +85,18 @@ def mk_full_signature_from_json(abi):
         for a in func['inputs']:
             arg = ast.arg(
```

Comment:
Add a note describing the issue that you found with ABI inputs that use `bytes` or `strings` and how the addition of an expected solves it. I have a feeling this brittleness will become an issue in future iterations, and it will be good to have an explanation to this effect somewhere in the code.